config/config/wc-config.yaml

```diff
@@ -144,10 +152,19 @@ fluentd:
     - effect: NoSchedule
       key: node-role.kubernetes.io/master
       value: ""
+    - key: node-role.kubernetes.io/control-plane
+      effect: NoSchedule
+      value: ""
 
   ## Only run on master nodes.
-  nodeSelector:
-    node-role.kubernetes.io/master: ""
+  nodeSelector: ""
+  affinity:
+    nodeAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+        nodeSelectorTerms:
+          - matchExpressions:
+              - key: node-role.kubernetes.io/master
+                operator: Exists
```

Comment:
Do you know if adding the `control-plane` thing here will make this break? Like is the `matchExpressions` an **and** or an **or**

Reply:
I think it would be rather safe to just have `control-plane` here, I think that label was added in all clusters starting with kuberentes v1.20
https://kubernetes.io/docs/reference/labels-annotations-taints/#node-role-kubernetes-io-master

Reply:
Sure, that makes sense to do.

But I was also curious if someone knew if it works like an and or an or.

Reply:
Looks like the list under nodeSelectorTerms is ORed. So it should be ok as long as one of them matches.
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#nodeselector-v1-core

Reply:
Then you can do AND logic inside matchExpressions and matchFields inside one of the nodeSelectorTerms

Reply:
Right, nice find!

Reply:
do we want the control-plane key inside the matchExpressions?